sql/core/src/main/scala/org/apache/spark/sql/UDFRegistration.scala

```diff
@@ -87,6 +87,7 @@ class UDFRegistration private[sql] (sqlContext: SQLContext) extends Logging {
     (0 to 22).map { x =>
       val types = (1 to x).foldRight("RT")((i, s) => {s"A$i, $s"})
       val typeTags = (1 to x).map(i => s"A${i}: TypeTag").foldLeft("RT: TypeTag")(_ + ", " + _)
+      val inputTypes = (1 to x).foldLeft("Nil")((s, i) => {s"$s :+ ScalaReflection.schemaFor[A$i].dataType"})
```

Comment:
I think the convention is usually
```scala
a :: b :: c :: Nil
```

rather than 
```scala
Nil :+ a :+ ...
```

Do you mind updating it?